MushroomObserver/mushroom-observer
# Create Zip file for DwCA export option

To test:
- Go to somewhere you can do an observation list export (e.g., go to a species list page then click 'Downloads, Reports, Print Labels')
- Select 'Darwin Core' and click Download

Expect result: A zip archive that can be successfully loaded into the GBIF data validator (https://www.gbif.org/tools/data-validator) without any modification.

Previous result: A CSV file that had to be combined with a meta.xml file and then bundled into a zip archive.

Changes: Added rubyzip gem.  Refactored ObservationReport:Base to separate out the table export specific stuff (BaseTable) from file export stuff (Base).  BaseTable inherits from Base and CSV now inherits from BaseTable.  Created DarwinCsv which is essentially the same as the old Darwin class.  Introduced ZipReport class which inherits from Base for creating Zip Archives.  Changed the Darwin class to inherit from ZipReport and uses DarwinCsv to build a real DwCA file.  Added a static meta.xml file.  Added a test and fixed up the existing tests.
